### ISSUE TITLE
A large amount of Pending pods waiting for resources seems to impact a new pod for which resources are available

### ISSUE THREAD

**Issue by Timvissers** [kubernetes/kubernetes#88108]
**What happened**:
Context: A kubernetes cluster with 2 custom namespaces: 'backend' and 'jobs'. 
- backend namespace contains up to 200 microservices pods.
- jobs namespace contains processing jobs/pods. These are heavy jobs that might take an hour of processing. There are added by incoming events.

Now, it can occur that suddenly 3000 events come in, resulting in 3000 kubernetes **jobs** being created. In our current setup, let's say a job requires cpu/memory resources somewhat less than a particular type of node (eg m5.large in aws ec2). We allow scaling up nodes up to a max of 120. What happens: 120 jobs start, their pods get status Running. The other 2880 pods have status Pending, they will have to wait for nodes to become available after they processed other jobs. This processing takes many hours, but works. 

In this situation, let's say a pod in the **backend** namespace is created, eg we manually delete a pod, and therefor a new one is created. The problem is that now this pod remains Pending for a couple of minutes before it becomes Running. This is unwanted behaviour.
I'm not talking about docker image pulling time or no resources available. Resources are available (pod capacity, memory, cpu, ...). We have different node groups for the backend pods versus the job pods!
When I do a 'kubectl describe pod' for this pod, there is no event (pulling, pulled, created, started, not even scheduled).

So it seems, this pod is impacted by the other load, in the other namespace.

**What you expected to happen**:
We would love the backend Pending pod to become available immediately, and not have impact by these other 2880 pending job pods.

**How to reproduce it (as minimally and precisely as possible)**:
See above: 
- generate a large number (couple of 1000s) of pending pods in node group 1, where they remain pending since there is not enough resources to handle them all.
- start a pod in node group 2, where there are enough resources to have the pod started.
- the latter pod will remain in Pending state for minutes, then becomes Running.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): EKS 1.13
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-14T04:24:29Z", GoVersion:"go1.12.13", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.12-eks-eb1860", GitCommit:"eb1860579253bb5bf83a5a03eb0330307ae26d18", GitTreeState:"clean", BuildDate:"2019-12-23T08:58:45Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration: EKS
- OS (e.g: `cat /etc/os-release`): amazon linux 2
- Kernel (e.g. `uname -a`): 4.14.133-113.112.amzn2.x86_64 #1 SMP Tue Jul 30 18:29:50 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Network plugin and version (if this is a network-related bug): aws cni 1.5.3
- Others:

**Comment by ahg-g** [comment]
If all the pods have the same priority, then the scheduling is near FIFO. Unschedulable pods will be retried according to an exponential backoff heuristic, so some of those unschedulable pods are consuming scheduling cycles and hence blocking the schedulable pod from getting scheduled as fast.

Two things we did recently that should improve your situation:

1) The scheduler performance improved dramatically since 1.13, and so I expect that on newer versions those unschedulable pods will be examined a lot faster. 
2) We recently fixed a bug in the backoff logic that resulted in unschedulable pods gettting retried sooner than they should: https://github.com/kubernetes/kubernetes/pull/87616, the fix was back ported to 1.16.
